framework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -0,0 +1,202 @@
+      algoResidual = algo.getResidual(signal, params, pivots, None) # TODO randomEngine
+      residual[0, :, indices] = algoResidual.T # transpose, again because of indices
+      # TODO meta store signal, residual?
+
+  def __evaluateLocal__(self, featureVals):
+    """
+      @ In, featureVals, float, a scalar feature value is passed as scaling factor
+      @ Out, rlz, dict, realization dictionary of values for each target
+    """
+    pivots = self.pivotParameterValues
+    result = np.zeros((self.pivotParameterValues.size, len(self.target) - 1)) # -1 is pivot
+    for algo in self.tsaAlgorithms:
+      targets = algo.target
+      indices = tuple(self.target.index(t) for t in targets)
+      params = self.trainedParams[algo]
+      signal = algo.generate(params, pivots, None) # TODO random engine for third argument
+      result[:, indices] = signal
+    # RAVEN realization construction
+    rlz = dict((target, signal[:, t]) for t, target in enumerate(self.target) if target != self.pivotParameterID)
+    rlz[self.pivotParameterID] = self.pivotParameterValues
+    return rlz
+
+  def writePointwiseData(self, writeTo):
+    """
+      Writes pointwise data about this ROM to the data object.
+      @ In, writeTo, DataObject, data structure into which data should be written
+      @ Out, None
+    """
+    pass # TODO
+
+  def writeXML(self, writeTo, targets=None, skip=None):
+    """
+      Allows the SVE to put whatever it wants into an XML to print to file.
+      Overload in subclasses.
+      @ In, writeTo, xmlUtils.StaticXmlElement, entity to write to
+      @ In, targets, list, optional, unused (kept for compatability)
+      @ In, skip, list, optional, unused (kept for compatability)
+      @ Out, None
+    """
+    root = writeTo.getRoot()
+    for algo in self.tsaAlgorithms:
+      algoNode = xmlUtils.newNode(algo.name)
+      algo.writeXML(algoNode, self.trainedParams[algo])
+      root.append(algoNode)
+
+  ### Segmenting and Clustering ###
+  def isClusterable(self):
+    """
+      Allows ROM to declare whether it has methods for clustring. Default is no.
+      @ In, None
+      @ Out, isClusterable, bool, if True then has clustering mechanics.
+    """
+    # clustering methods have been added
+    return False # TODO
+
+  ### ESSENTIALLY UNUSED ###
+  def _localNormalizeData(self,values,names,feat):
+    """
+      Overwrites default normalization procedure, since we do not desire normalization in this implementation.
+      @ In, values, unused
+      @ In, names, unused
+      @ In, feat, feature to normalize
+      @ Out, None
+    """
+    self.muAndSigmaFeatures[feat] = (0.0,1.0)
+
+  def __confidenceLocal__(self,featureVals):
+    """
+      This method is currently not needed for ARMA
+    """
+    pass
+
+  def __resetLocal__(self,featureVals):
+    """
+      After this method the ROM should be described only by the initial parameter settings
```

Comment:
I'm not 100% sure what you mean, unless it's just grammar checking; since this is the same comment left on these same methods in many of the SVL, I think it might be better to maintain the consistency. We can create an issue to track the fix across SVL if you want.

Reply:
I was just commenting on some of the minor things missing like periods, and capitalizing the first word in the sentence. I looked at the other methods in the file and they used periods and such so it was just a comment for consistency.